evm/vm/flavors/tester/__init__.py

```diff
@@ -0,0 +1,121 @@
+from cytoolz import (
+    assoc,
+)
+
+from eth_utils import (
+    reversed_return,
+)
+
+from evm import (
+    Chain,
+)
+
+from evm.vm.flavors import (
+    FrontierVM as BaseFrontierVM,
+    HomesteadVM as BaseHomesteadVM,
+    EIP150VM as BaseEIP150VM,
+)
+
+from evm.utils.chain import (
+    generate_vms_by_range,
+)
+
+
+def chfp_method_factory(parent_class):
+    def create_tester_header_from_parent(vm_class, parent_header, **header_params):
+        """
+        Creates and initializes a new block header from the provided
+        `parent_header`.
+        """
+        return parent_class.create_header_from_parent(
+            parent_header,
+            **assoc(header_params, 'gas_limit', parent_header.gas_limit)
+        )
+    return create_tester_header_from_parent
+
+
+FrontierTesterVM = BaseFrontierVM.configure(
+    name='FrontierTesterVM',
+    create_header_from_parent=classmethod(chfp_method_factory(BaseFrontierVM)),
+)
+
+
+BaseHomesteadTesterVM = BaseHomesteadVM.configure(
+    name='HomesteadTesterVM',
+    create_header_from_parent=classmethod(chfp_method_factory(BaseHomesteadVM)),
+)
+
+
+EIP150TesterVM = BaseEIP150VM.configure(
+    name='EIP150TesterVM',
+    create_header_from_parent=classmethod(chfp_method_factory(BaseEIP150VM)),
+)
+
+
+INVALID_FORK_ACTIVATION_MSG = (
+    "The {0}-fork activation block may not be null if the {1}-fork block "
+    "is non null"
+)
+
+
+@reversed_return
+def _generate_vm_configuration(homestead=None, dao=None, anti_dos=None):
```

Comment:
It took me a while to realize what values the arguments here could take. Maybe add a _start_block suffix to them, or a docstring?